File: crates/quench-node/polyfills/post-bootstrap/process-surface-15.js
{
  if (globalThis.process && globalThis.process.stdout) {
    const stdout = globalThis.process.stdout;
    stdout.writableHighWaterMark = 65536;
    if (stdout.constructor.name !== "Socket") {
      Object.defineProperty(stdout, "constructor", {
        value: function Socket() {},
        configurable: true
      });
    }
  }
}

File: crates/quench-node/src/polyfills/post_bootstrap/process_surface_15.rs
//! Polyfill: `process-surface-15`

pub const JS: &str = quench_js_check::checked_js!(r#"if (globalThis.process) {
  for (const stream of [globalThis.process.stdout, globalThis.process.stderr]) {
    if (!stream) continue;
    stream.writableHighWaterMark = 65536;
    if (stream.constructor.name !== "Socket") {
      Object.defineProperty(stream, "constructor", {
        value: function Socket() {},
        configurable: true
      });
    }
  }
}
"#);
